app/controllers/case_logs_controller.rb

```diff
@@ -107,9 +107,13 @@ def responses_for_page(page)
         day = params["case_log"]["#{question_key}(3i)"]
         month = params["case_log"]["#{question_key}(2i)"]
         year = params["case_log"]["#{question_key}(1i)"]
-        next unless day.present? && month.present? && year.present?
+        next unless [day, month, year].any?(&:present?)
 
-        result[question_key] = Date.new(year.to_i, month.to_i, day.to_i)
+        result[question_key] = if day.to_i.between?(1, 31) && month.to_i.between?(1, 12) && year.to_i.between?(2000, 2200)
+                                 Date.new(year.to_i, month.to_i, day.to_i)
+                               else
+                                 Date.new(0, 1, 1)
```

Comment:
this is a bit hacky, but I couldn't figure out a way to indicate that the attempt to fill in the date happened vs users just ignoring the question

Reply:
What about saving the `day, month, year` fields here and then deriving the datetime in the validation?

Reply:
I thought about that, it would work for mrcdate, but for other dates I'm not sure if we're partitioning them into separate fields

Reply:
Fair enough. Would be nice to have a consistent way to treat them I suppose but happy for this to go in first anyway.